unity-renderer/Assets/UIComponents/Scripts/Components/DynamicIconList/DynamicListComponentView.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class DynamicListComponentView : BaseComponentView, IDynamicListComponentView
+{
+    [SerializeField] private Image iconReference;
+
+    internal List<Image> instantiatedIcons = new ();
+    public override void RefreshControl()
+    {
+    }
+
+    public void AddIcon(Sprite sprite)
+    {
+        Image newImage = Instantiate(iconReference, transform);
+        newImage.sprite = sprite;
+        instantiatedIcons.Add(newImage);
+    }
+
+    public void AddIcons(List<Sprite> spriteList)
+    {
+        RemoveIcons();
+
+        foreach (var sprite in spriteList)
+        {
+            AddIcon(sprite);
+        }
+    }
+
+    public void RemoveIcons()
+    {
+        foreach (var icon in instantiatedIcons)
+            DestroyImmediate(icon.gameObject);
```

Comment:
why `DestroyImmediate`? it is not recommended in runtime

Reply:
ah true, I think I did it for the tests, my bad